### PR TITLE
FEAT: Add oembed for assets

### LINKED ISSUE #392: Asset Database Discourse Onebox
Presumably this is implemented Discourse side rather than PyRIGS side but still makes sense on this tracker given its a feature for this.

### PR DESCRIPTION
**Pull request by FreneticScribbler**
That was easy!

Don't see the worth in doing supplier currently...we don't do Org/Venue etc after all...

Will close #392

### CODE REVIEW

**Comment by FreneticScribbler** [comment]
> That was easy!

Famous last words...

**Comment by FreneticScribbler** [comment]
Should work now, https://oembed.frdnspnzr.de/ reckons it does but the forum is still sad. Does it need to be whitelisted or smth?

File: assets/urls.py
Comment (mattysmith22) [inline assets/urls.py:14]: Why the difference from how oembed is done for RIGS/urls.py? I understand you need a permission for that, but surely you could use the asset_view perm?

Just trying to ensure consistency

**Review by mattysmith22 — COMMENTED** [review]
Great! Just want to know why the difference in usage between assets and RIGS, other than that I am happy

File: assets/urls.py
Comment (FreneticScribbler) [reply assets/urls.py:14]: We are currently (ab)using asset_view to be a 'view *ALL* information about an asset' rather than a be able to view assets perm. Basic users can see assets without permissions, the view_asset perm is unused in favour of 'view_asset_financial' AFAIK. 

Why not add the 'view_asset' to all users and require it? Because that's the same thing as requiring no perm, but with more steps?  

When I open up RIGS to reveal more information to basic users, the behaviour will be consistent in not requiring permissions for either oembed.

Comment (mattysmith22) [reply assets/urls.py:14]: Surely though the whole point of permissions is that they can be created and revoked? If the permission exists, we should honour it on the backend so that in the future if an admin wants to create a set of users that don't have access and just revoke the permission without checking that it actually occurs, we don't have a breach?

Personally I think we should have this granular approach, but if you have thought about it and see a reason why not, then that is fine and I'll approve it

Comment (FreneticScribbler) [reply assets/urls.py:14]: In a general world situation, totally get you. I think in the case of RIGS I think the usecase is sufficiently different to warrant this. In what possible situation might we need such a group? 

I do recognise it is a bit misleading, but the current usecase would be that these permissions are on all users, which is tedious to apply...

Comment (FreneticScribbler) [reply assets/urls.py:14]: Perhaps the solution is to disable/remove the view asset perm, if that's possible?

Comment (mattysmith22) [reply assets/urls.py:14]: That would make sense, though tbh is as much effort as just doing it the same way as in rigs! The only reason I brought it up as well that we can have one way or doing oembeds, rather than two

https://stackoverflow.com/questions/6062655/remove-or-hide-default-permissions-from-django

Comment (FreneticScribbler) [reply assets/urls.py:14]: Well like I say once my other PR is also merged it will be unified. The reality is its unlikely that future maintainers will be without support from us, and if they are then they're probably doing a full rewrite anyway :P

Comment (FreneticScribbler) [reply assets/urls.py:14]: Happy to merge?

Comment (mattysmith22) [reply assets/urls.py:14]: Yeah, approving the merge now

**Review by mattysmith22 — APPROVED** [review]

**Pull request merged**
